### ISSUE TITLE
Example sampling from vine copula with conditioned variables

### ISSUE THREAD

**Issue by MattDerry** [vinecopulib/pyvinecopulib#86]
I'm quite new to this type of model and still wrapping my head around the details, but I've run across this library and found it very helpful (thanks!).  

I have a problem where I would like to fit a model to 70k sample of 11 dimensional data.  Once fit, I'd love to be able to do subsequent simulations where I condition on 9 dimensions of the data (which are known) and then sample from the model on the  remaining 2 dimensions given that conditioning.  Ultimately I'm looking for time series of those 2 dimensions that ideally models the statistical properties of the initial distribution of time series that the model was fit on.  Is this something that can be done with the current API?

Thank you again!

**Comment by tnagler** [comment]
Yes this is possible with the current API using the Rosenblatt transform and its inverse. The approach is explained in Section 4.2 of [this paper](https://arxiv.org/pdf/2102.06416.pdf). 

In essence, you need to ensure that the tree structure of your model is suitable for conditional simulation. More specifically, the two variables you want to simulate must appear on the bottom left of the diagonal of the R-vine array. In the paper above, this is ensured by creating a suitable [DVineStructure](https://vinecopulib.github.io/pyvinecopulib/_generate/pyvinecopulib.DVineStructure.html) and fitting the model on this structure. The remaining steps can be implemented using  [Vinecop.rosenblatt()](https://vinecopulib.github.io/pyvinecopulib/_generate/pyvinecopulib.Vinecop.rosenblatt.html) and [Vinecop.inverse_rosenblatt()](https://vinecopulib.github.io/pyvinecopulib/_generate/pyvinecopulib.Vinecop.inverse_rosenblatt.html).

**Comment by couasnonanais** [comment]
Also very happy to see robust developments of Python packages for building C/D/R Vine structures, many thanks!
Would it be possible to add a simplified example of the approach you are mentioning above @tnagler ? This would be very helpful.

**Comment by lassiterdc** [comment]
Are there any updates on this front? I'm trying to do something very similar to @MattDerry and am not finding any existing libraries with easy-to-use functions for conditional simulation from a vine copula. I'm hoping pyvinecopulib could be the one!

**Comment by couasnonanais** [comment]
Maybe nice to mention this package that provide direct functions for conditional simulation: https://vinecopulas.readthedocs.io/en/latest/index.html even though I have not tested them myself yet

**Comment by markcus97** [comment]
There's a recent paper which implements conditional sampling from an arbitrarily structured simplified vine using MCMC: Hanebeck, A., Şahin, Ö., Havlíčková, P. et al. Sampling from Conditional Distributions of Simplified Vines. Stat Comput 35, 128 (2025). [https://doi.org/10.1007/s11222-025-10652-4](https://doi.org/10.1007/s11222-025-10652-4)

I translated their R code to python and wrote an easy-to-use interface in this library: [https://github.com/markcus97/condcopula](https://github.com/markcus97/condcopula). You pass a copula object (bivariate or vine) from pyvinecopulib and then you can conditionally sample that object.

Hope it's useful and let me know if you have any suggestions for improvements.
